code/modules/cargo/order.dm

```diff
@@ -86,6 +93,15 @@
 /datum/supply_order/proc/generate(atom/A)
 	var/obj/structure/closet/crate/C = pack.generate(A)
 	var/obj/item/paper/fluff/jobs/cargo/manifest/M = generateManifest(C)
+	
+	if(paying_account)

```

Comment:
Just pass this information into generate functions and modify it, this stacking thing is pretty ugly.

Reply:
Do you mean code-wise or the fact that the crate is contained within the privacy crate?

Reply:
Both.

Reply:
Oh, i see now, i don't know why i didn't think of that. I'll make it change the crate type then.

Reply:
Wait, i forgot that the reason 4 is to retain the crate's original access requirements as a separate lock; now cargo can order crates with its budget to freely unlock them